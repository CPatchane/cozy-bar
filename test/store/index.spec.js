import createStore from 'lib/store'
import {
  isAppListFetching,
  getApps,
  setInfos,
  hasFetched
} from 'lib/reducers'

describe('store', () => {
  let store

  beforeEach(() => {
    store = createStore()
  })

  const getState = () => store.getState()

  it('should keep fetching status', () => {
    expect(isAppListFetching(getState())).toBe(false)
    store.dispatch({ type: 'FETCH_APPS' })
    expect(isAppListFetching(getState())).toBe(true)
    store.dispatch({ type: 'RECEIVE_APP_LIST' })
    expect(isAppListFetching(getState())).toBe(false)
  })

  it('should keep apps data', () => {
    const apps = [{ slug: 'drive' }, {slug: 'banks'}]
    expect(getApps(getState())).toEqual([])
    store.dispatch({ type: 'RECEIVE_APP_LIST', apps: apps })
    expect(getApps(getState())).toEqual(apps)
  })

  it('should remember it has fetched (ok case)', () => {
    expect(hasFetched(getState())).toEqual(false)
    store.dispatch({ type: 'RECEIVE_APP_LIST' })
    expect(hasFetched(getState())).toEqual(true)
  })

  it('should remember it has fetched (bad case)', () => {
    expect(hasFetched(getState())).toEqual(false)
    store.dispatch({ type: 'RECEIVE_APP_LIST_FORBIDDEN' })
    expect(hasFetched(getState())).toEqual(false)
  })

  it('should not return current app on mobile', () => {
    const apps = [
      { slug: 'drive', appName: 'Cozy Drive', comingSoon: true, name_prefix: 'cozydrive://' },
      {slug: 'banks'}
    ]
    const onMobile = true
    store.dispatch(setInfos('Cozy Drive', 'cozydrive://'))
    expect(getApps(getState(), onMobile)).toEqual([])
    store.dispatch({ type: 'RECEIVE_APP_LIST', apps: apps })
    expect(getApps(getState(), onMobile)).toEqual([apps[1]])
  })
})
