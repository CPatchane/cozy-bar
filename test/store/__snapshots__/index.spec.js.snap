// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store should keep apps data 1`] = `
Array [
  Object {
    "href": undefined,
    "isCurrentApp": false,
    "slug": "drive",
  },
  Object {
    "href": undefined,
    "isCurrentApp": false,
    "slug": "banks",
  },
]
`;
